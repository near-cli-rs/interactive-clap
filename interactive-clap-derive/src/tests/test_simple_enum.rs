use super::pretty_codegen;

#[test]
fn test_simple_enum() {
    let input = syn::parse_quote! {
        pub enum Mode {
            /// Prepare and, optionally, submit a new transaction with online mode
            Network,
            /// Prepare and, optionally, submit a new transaction with offline mode
            Offline,
        }
    };

    let interactive_clap_codegen = crate::derives::interactive_clap::impl_interactive_clap(&input);
    insta::assert_snapshot!(pretty_codegen(&interactive_clap_codegen));

    let step_one_output = syn::parse_quote! {
        pub enum CliMode {
            /// Prepare and, optionally, submit a new transaction with online mode
            Network,
            /// Prepare and, optionally, submit a new transaction with offline mode
            Offline,
        }
    };

    let to_cli_args_codegen = crate::derives::to_cli_args::impl_to_cli_args(&step_one_output);
    insta::assert_snapshot!(pretty_codegen(&to_cli_args_codegen));
}

#[test]
fn test_simple_enum_with_strum_discriminants() {
    let input = syn::parse_quote! {
        #[strum_discriminants(derive(EnumMessage, EnumIter))]
        /// A little beautiful comment about our choice
        pub enum Mode {
            /// Prepare and, optionally, submit a new transaction with online mode
            #[strum_discriminants(strum(message = "Yes, I keep it simple"))]
            Network,
            /// Prepare and, optionally, submit a new transaction with offline mode
            #[strum_discriminants(strum(
                message = "No, I want to work in no-network (air-gapped) environment"
            ))]
            Offline,
        }
    };

    let interactive_clap_codegen = crate::derives::interactive_clap::impl_interactive_clap(&input);
    insta::assert_snapshot!(pretty_codegen(&interactive_clap_codegen));

    let step_one_output = syn::parse_quote! {
        pub enum CliMode {
            /// Prepare and, optionally, submit a new transaction with online mode
            Network,
            /// Prepare and, optionally, submit a new transaction with offline mode
            Offline,
        }
    };

    let to_cli_args_codegen = crate::derives::to_cli_args::impl_to_cli_args(&step_one_output);
    insta::assert_snapshot!(pretty_codegen(&to_cli_args_codegen));
}
