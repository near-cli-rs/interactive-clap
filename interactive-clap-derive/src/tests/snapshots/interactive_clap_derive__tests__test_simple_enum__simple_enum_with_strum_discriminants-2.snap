---
source: interactive-clap-derive/src/tests/test_simple_enum.rs
expression: pretty_codegen(&to_cli_args_codegen)
---
impl interactive_clap::ToCliArgs for CliMode {
    fn to_cli_args(&self) -> std::collections::VecDeque<String> {
        match self {
            Self::Network => {
                let mut args = std::collections::VecDeque::new();
                args.push_front("network".to_owned());
                args
            }
            Self::Offline => {
                let mut args = std::collections::VecDeque::new();
                args.push_front("offline".to_owned());
                args
            }
        }
    }
}
